File: solutions/javascript/raindrops/raindrops.js
export const convert = (num) => {
  const out = []
  for (const [factor, sound] of [[3, 'Pling'], [5, 'Plang'], [7, 'Plong']]) {
    if (num % factor == 0) {
      out.push(sound)
    }
  }
  return out.join("") || `${num}`
};

File: solutions/ruby/raindrops/raindrops.rb
module Raindrops
  SOUNDS = { 3 => 'Pling', 5 => 'Plang', 7 => 'Plong' }.freeze

  def self.convert(number)
    sound = SOUNDS.select { |factor, _| (number % factor).zero? }.values.join('')
    sound.empty? ? number.to_s : sound
  end
end
